src/math/polyfit/mod.rs

```diff
@@ -0,0 +1,42 @@
+/*
+ * ANISE Toolkit
+ * Copyright (C) 2021-2022 Christopher Rabotin <christopher.rabotin@gmail.com> et al. (cf. AUTHORS.md)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Documentation: https://nyxspace.com/
+ */
+
+use core::ops::{Index, IndexMut};
+
+/// A FixedArray is a a way around allocating vectors when we don't know the exact size at compile time.
+/// This will be made obsolete when https://github.com/rust-lang/rust/issues/60551 is merged into rust stable.
+#[derive(Copy, Clone, Debug)]
+pub struct F64TupleArray<const N: usize, const S: usize>([[f64; N]; S]);
```

Comment:
Keeping this for posteriori to enable building a Hermite interpolation scheme without the intermediate array that contains the ordinates and first derivatives (abscissas / epochs are contiguous so we don't store them in an array).

A struct of an underlying `&'a [f64]` can be used to skip items such that the indexing is every sixth element for example.